src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java

```diff
@@ -0,0 +1,111 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.modifications;
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.SwitchKind;
+import lombok.SneakyThrows;
+import org.gridsuite.modification.server.ModificationType;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.BusbarConnectionCreationInfos;
+import org.gridsuite.modification.server.dto.ModificationInfos;
+import org.gridsuite.modification.server.dto.VoltageLevelCreationInfos;
+import org.gridsuite.modification.server.utils.MatcherModificationInfos;
+import org.gridsuite.modification.server.utils.MatcherVoltageLevelCreationInfos;
+import org.gridsuite.modification.server.utils.ModificationCreation;
+import org.gridsuite.modification.server.utils.NetworkCreation;
+import org.junit.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.*;
+import static org.gridsuite.modification.server.NetworkModificationException.Type.CREATE_VOLTAGE_LEVEL_ERROR;
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author walid Sahnoun <walid.sahnoun at rte-france.com>
+ */
+public class VoltageLevelCreationTest extends AbstractNetworkModificationTest {
+
+    @Override
+    protected Network createNetwork(UUID networkUuid) {
+        return NetworkCreation.create(networkUuid, true);
+    }
+
+    @Override
+    protected ModificationInfos buildModification() {
+        return ModificationCreation.getCreationVoltageLevel("s2", "vlId", "vlName");
+    }
+
+    @Override
+    protected ModificationInfos buildModificationUpdate() {
+        return VoltageLevelCreationInfos.builder()
+                .equipmentId("VoltageLevelIdEdited")
+                .equipmentName("VoltageLevelEdited")
+                .nominalVoltage(385)
+                .substationId("s2")
+                .busbarSections(List.of())
+                .busbarConnections(List.of())
+                .type(ModificationType.VOLTAGE_LEVEL_CREATION)
+                .build();
+    }
+
+    @Override
+    protected MatcherModificationInfos createMatcher(ModificationInfos modificationInfos) {
+        return MatcherVoltageLevelCreationInfos.createMatcherVoltageLevelCreationInfos((VoltageLevelCreationInfos) modificationInfos);
+    }
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertNotNull(getNetwork().getVoltageLevel("vlId"));
+    }
```

Comment:
you can go deeper in network assertions, adding:
```
assertNotNull(getNetwork().getBusbarSection("bbs.nw"));
assertNotNull(getNetwork().getBusbarSection("bbs.ne"));
assertNotNull(getNetwork().getBusbarSection("bbs.sw"));
assertTrue(getNetwork().getSubstation("s2").getVoltageLevelStream().anyMatch(vl -> vl.getId().equals("vlId")));
```

Reply:
done